framework/src/media/media_player.c

```diff
@@ -0,0 +1,336 @@
+/****************************************************************************
+ *
+ * Copyright 2017 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#include <media/media_player.h>
+#include <pthread.h>
+
+#define IS_SOCKET(fd) (fd > CONFIG_NFILE_DESCRIPTORS)
+#define PLAY_LOCK() do { pthread_mutex_lock(&g_pc.mutex_play); } while (false)
+#define PLAY_UNLOCK() do { pthread_mutex_unlock(&g_pc.mutex_play); } while (false)
+
+#define ST_PROTO_PLAY	0x00
+#define ST_PROTO_STOP	0x01
+#define ST_PROTO_PAUSE	0x02
+#define ST_PROTO_RESUME	0x03
+
+struct media_play_context_s {
+	pthread_t           pth;
+	bool                thread_running;
+	pthread_mutexattr_t mutex_attr_play;
+	pthread_mutex_t     mutex_play;
+	sem_t               sem_pause;
+	struct pcm         *pcmout;
+	unsigned int        buffer_size;
+	int                 fd;
+	int		    download_size;
+	media_format_e      format;
+	play_state_t        state;
+};
+
+static struct media_play_context_s g_pc;
+int media_playing(void *args);
+
+/**
+ * Public Functions
+ */
+play_result_t media_play(int fd, media_format_e format)
+{
+	g_pc.fd = fd;
+
+	struct pcm_config config;
```

Comment:
Let's add TODO here.
We will use parse_header in future.
parse_header(&config);
Also, basic config will be set in tinyalsa layer if there's no use defined config value.
You can remove below config setting.